### PR TITLE
Support default dtype in `L.GoogLeNet`

### PR DESCRIPTION
**Pull request by takagi**
This PR is a part of #4582, makes `L.GoogLeNet` use the default dtype in its methods.

### CODE REVIEW

**Comment by okuta** [comment]
jenkins, test this please.

File: tests/chainer_tests/links_tests/model_tests/test_vision.py [hunk 1]
```diff
@@ -259,13 +260,28 @@ def test_copy_gpu(self):
...
 
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32],
+}))
 @unittest.skipUnless(googlenet.available, 'Pillow is required')
 @attr.slow
 class TestGoogLeNet(unittest.TestCase):
 
     def setUp(self):
+        self._old_dtype = None
```
Comment (okuta) [inline tests/chainer_tests/links_tests/model_tests/test_vision.py]: Maybe you can use `getattr`.
```self._old_dtype = getattr(config._local, 'dtype', None)```

Comment (takagi) [reply tests/chainer_tests/links_tests/model_tests/test_vision.py]: Fixed!

File: tests/chainer_tests/links_tests/model_tests/test_vision.py [hunk 2]
```diff
@@ -259,13 +260,26 @@ def test_copy_gpu(self):
...
 
+@testing.parameterize(*testing.product({
+    'dtype': [numpy.float16, numpy.float32],
+}))
 @unittest.skipUnless(googlenet.available, 'Pillow is required')
 @attr.slow
 class TestGoogLeNet(unittest.TestCase):
 
     def setUp(self):
+        config = chainer.config
```
Comment (beam2d) [inline tests/chainer_tests/links_tests/model_tests/test_vision.py]: See this comment: https://github.com/chainer/chainer/pull/5114#discussion_r203917883

Comment (takagi) [reply tests/chainer_tests/links_tests/model_tests/test_vision.py]: I need to be good at Python more :(

Comment (takagi) [reply tests/chainer_tests/links_tests/model_tests/test_vision.py]: Fixed!

**Comment by takagi** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**